[OSDOCS-11350]: Adding more versioning details to HCP docs (#84335)
openshift/openshift-docs: modules/hosted-control-planes-version-support.adoc

[id="hosted-control-planes-version-support_{context}"]
= Versioning for {hcp}

> The {hcp} feature includes the following components, which might require independent versioning an support levels:

Reviewer: Guessing on the typo fix:
```suggestion
The {hcp} feature includes the following components, which might require independent versioning and support levels:
```

Author: Good catch!

* Management cluster
* HyperShift Operator
* {hcp-capital} (`hcp`) command-line interface (CLI)
* `hypershift.openshift.io` API
* Control Plane Operator

[id="hcp-versioning-mgmt_{context}"]
== Management cluster

> In management clusters for production use, you need {mce}, which is available through OperatorHub. The {mce-short} bundles a supported build of the HyperShift Operator. For your management clusters to remain supported, you must use version of {product-title} that {mce-short} runs on. In general, a new release of {mce-short} runs on the following versions of {product-title}:

Reviewer: Maybe?
```suggestion
In management clusters for production use, you need {mce}, which is available through OperatorHub. The {mce-short} bundles a supported build of the HyperShift Operator. For your management clusters to remain supported, you must use the version of {product-title} that {mce-short} runs on. In general, a new release of {mce-short} runs on the following versions of {product-title}:
```

Author: Another good catch.

> * The latest Generally Available version of {product-title}
> * Two versions before the latest Generally Available version of {product-title}

Reviewer: I don't see that we have explicit guidelines related to this anywhere, but, I'm thinking about whether this should be lowercase "generally available" (as a sort of general reference) vs "General Availability" (which is what I have typically known "GA" to specifically stand for, as a "Common Release Term" / release level).

I am sure I am just splitting hairs here. 😅 And FWIW, we definitely are not consistent on this across the repo (can find examples backing up either). 🤷 

So, just a suggestion:
```suggestion
* The latest General Availability version of {product-title}
* Two versions before the latest General Availability version of {product-title}
```

Author: I think you are right -- GA means General Availability. I'll make those changes.

> The full list of {product-title} versions that you can install through the HyperShift Operator on a management cluster depends on the version of your HyperShift Operator. However, the list always includes at least the same {product-title} version as the management cluster and two previous minor versions relative to hte management cluster. For example, if the management cluster is running 4.17 and a supported version of {mce-short}, the HyperShift Operator can install 4.17, 4.16, and 4.15 hosted clusters.

Reviewer 2: ```suggestion
The full list of {product-title} versions that you can install through the HyperShift Operator on a management cluster depends on the version of your HyperShift Operator. However, the list always includes at least the same {product-title} version as the management cluster and two previous minor versions relative to the management cluster. For example, if the management cluster is running 4.17 and a supported version of {mce-short}, the HyperShift Operator can install 4.17, 4.16, and 4.15 hosted clusters.
```

> The full list of {product-title} versions that you can install through the HyperShift Operator on a management cluster depends on the version of your HyperShift Operator. However, the list always includes at least the same {product-title} version as the management cluster and two previous minor versions relative to hte management cluster. For example, if the management cluster is running 4.17 and a supported version of {mce-short}, the HyperShift Operator can install 4.17, 4.16, and 4.15 hosted clusters.

Reviewer 3: 4.17, 4.16, 4.15, and 4.14 hosted clusters

Reviewer 4: > two previous minor versions relative to the management cluster. For example, if the management cluster is running 4.17 and a supported version of multicluster engine Operator, the HyperShift Operator can install 4.17, 4.16, 4.15, and 4.14 hosted clusters.

It may be confusing for the customers, as they are not in an N-2 relationship (4.14 hc for 4.17 HO)?

Author: Updating to 4.14-4.17 per conversation with Adel.

----

[id="hcp-versioning-cli_{context}"]
== {hcp} CLI

You can use the `hcp` CLI to create hosted clusters. You can download the CLI from {mce-short}. When you run the `hcp version` command, the output shows the latest {product-title} that the CLI supports against your `kubeconfig` file.

[id="hcp-versioning-api_{context}"]
== hypershift.openshift.io API

You can use the `hypershift.openshift.io` API resources, such as, `HostedCluster` and `NodePool`, to create and manage {product-title} clusters at scale. A `HostedCluster` resource contains the control plane and common data plane configuration. When you create a `HostedCluster` resource, you have a fully functional control plane with no attached nodes. A `NodePool` resource is a scalable set of worker nodes that is attached to a `HostedCluster` resource.

The API version policy generally aligns with the policy for link:https://kubernetes.io/docs/reference/using-api/#api-versioning[Kubernetes API versioning].

Updates for {hcp} involve updating the hosted cluster and the node pools. For more information, see "Updates for hosted control planes".

[id="hcp-versioning-cpo_{context}"]
== Control Plane Operator

> The Control Plane Operator is released as part of each {product-title} payload release image:

Reviewer: ```suggestion
The Control Plane Operator is released as part of each {product-title} payload release image for the following architectures:
```
or
```suggestion
The Control Plane Operator is released as part of the {product-title} payload release image for each architecture:
```
or similar? Probably also clear enough as is.

Author: I went with your first suggestion.
